src/libwalletqt/TransactionHistory.cpp

```diff
@@ -113,3 +114,68 @@ TransactionHistory::TransactionHistory(Monero::TransactionHistory *pimpl, QObjec
     m_firstDateTime  = QDateTime(QDate(2014, 4, 18)); // the genesis block
     m_lastDateTime = QDateTime::currentDateTime().addDays(1); // tomorrow (guard against jitter and timezones)
 }
+
+QString TransactionHistory::writeCSV(quint32 accountIndex, QString out)
+{
+    QList<TransactionInfo *> history = this->getAll(accountIndex);
+    if(history.count() < 1){
+        return QString("");
+    }
+
+    // construct filename
+    qint64 now = QDateTime::currentDateTime().currentMSecsSinceEpoch();
+    QString fn = QString(QString("%1/monero-txs_%2.csv").arg(out, QString::number(now / 1000)));
+
+    // open file
+    QFile data(fn);
+    if(!data.open(QFile::WriteOnly | QFile::Truncate)){
+        return QString("");
+    }
+
+    // write header
+    QTextStream output(&data);
+    output << "blockHeight,epoch,time,direction,amount,atomicAmount,fee,txid,label,subaddrAccount,paymentId\n";
+
+    foreach(const TransactionInfo *info, history)
+    {
+        // collect column data
+        double amount = info->amount();
+        quint64 atomicAmount = info->atomicAmount();
+        quint32 subaddrAccount = info->subaddrAccount();
+        QString fee = info->fee();
+        QString direction = QString("");
+        TransactionInfo::Direction _direction = info->direction();
+        if(_direction == TransactionInfo::Direction_In)
+        {
+            direction = QString("in");
+        }
+        else if(_direction == TransactionInfo::Direction_Out){
+            direction = QString("out");
+        }
+        else {
+            continue;  // skip TransactionInfo::Direction_Both
+        }
+        QString label = info->label();
+        label.remove(QChar('"'));  // reserved
```

Comment:
Also should handle extra commas in label.

add
label.remove(QChar(','));  // reserved

Reply:
Actually the fix is to enclose `%9` in double quotes, which I forgot to do it seems.